src/main/java/jenkinsci/plugins/influxdb/generators/JenkinsBasePointGenerator.java

```diff
@@ -112,7 +120,25 @@ public boolean hasReport() {
         }
 
         if (hasMetricsPlugin(build)) {
+            java.util.logging.Logger LOGGER = java.util.logging.Logger.getLogger(this.getClass().getName());
+
             point.addField(TIME_IN_QUEUE, build.getAction(jenkins.metrics.impl.TimeInQueueAction.class).getQueuingDurationMillis());
+            point.addField(TOTAL_TIME_IN_QUEUE, build.getAction(jenkins.metrics.impl.TimeInQueueAction.class).getQueuingTimeMillis());
+            point.addField(NUM_SUBTASKS, build.getAction(jenkins.metrics.impl.TimeInQueueAction.class).getSubTaskCount());
+
+            // Subtask Point Generator -- Used to check longest subtask wait time
+            int subtask_count = 0;
+            for (Map.Entry<String, Long> subtask :
+                build.getAction(jenkins.metrics.impl.TimeInQueueAction.class).getSubTaskMap().entrySet()) {
+
+                if (subtask_count > MAX_SUBTASKS) {
+                    LOGGER.log(Level.WARNING, "Too many subtasks");
```

Comment:
I would make the warning message more descriptive. Adding the job name and job run number would be nice, as well as the MAX_SUBTASKS threshold that was hit. 

It's best to imagine an admin looking through some random jenkins logs and seeing "Too many subtasks", they would want more context!

Reply:
Logger should now show job name, run number, and MAX_SUBTASKS threshold! Tested briefly on test-Jenkins server